editing/edit-with-branch-versioning/src/main/java/com/esri/samples/edit_with_branch_versioning/EditWithBranchVersioningController.java

```
@@ -0,0 +1,359 @@
+import javafx.scene.layout.VBox;
+
+import com.esri.arcgisruntime.arcgisservices.ServiceVersionInfo;
+import com.esri.arcgisruntime.arcgisservices.ServiceVersionParameters;
+import com.esri.arcgisruntime.arcgisservices.VersionAccess;
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.data.ArcGISFeature;
+import com.esri.arcgisruntime.data.FeatureTableEditResult;
+import com.esri.arcgisruntime.data.ServiceFeatureTable;
+import com.esri.arcgisruntime.data.ServiceGeodatabase;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.GeoElement;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+
+public class EditWithBranchVersioningController {
+
+  @FXML private Button createVersionButton;
+  @FXML private Button switchVersionButton;
+  @FXML private ComboBox<String> damageTypeComboBox;
+  @FXML private ComboBox<VersionAccess> accessTypeComboBox;
+  @FXML private Label currentVersionLabel;
+  @FXML private MapView mapView;
+  @FXML private ProgressIndicator progressIndicator;
+  @FXML private TextField descriptionTextField;
+  @FXML private TextField nameTextField;
+  @FXML private VBox createVersionVBox;
+  @FXML private VBox editFeatureVBox;
+
+  private ArcGISFeature selectedFeature;
+  private FeatureLayer featureLayer;
+  private ServiceFeatureTable serviceFeatureTable;
+  private ServiceGeodatabase serviceGeodatabase; // keeps loadable in scope to avoid garbage collection
+  private String defaultVersion;
```

can we rename this to `defaultVersionName`: currently the variable name is a little obscure as to what a `defaultVersion` object is, especially when reading it later down the code without the context. 
```suggestion
  private String defaultVersionName;
```